### PR TITLE
Update testcases: use of checkers on docker_cli_nat_test.go

### PR DESCRIPTION
**Pull request by liaoqingwei**
Signed-off-by: liaoqingwei liaoqingwei@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_nat_test.go [hunk 1]
```diff
@@ -105,7 +92,5 @@ func (s *DockerSuite) TestNetworkLoopbackNat(c *check.C) {
 	out, _ := dockerCmd(c, "run", "-t", "--net=container:server", "busybox",
 		"sh", "-c", fmt.Sprintf("stty raw && nc -w 5 %s 8080", endpoint.String()))
 	final := strings.TrimRight(string(out), "\n")
-	if final != msg {
-		c.Fatalf("Expected message %q but received %q", msg, final)
-	}
+	c.Assert(final, check.Equals, msg, check.Commentf("Expected message %q but received %q", msg, final))
```
Comment (HuKeping) [inline integration-cli/docker_cli_nat_test.go]: The error comment is nothing special so I think `c.Assert(final, checker.Equals, msg)` is OK.

File: integration-cli/docker_cli_nat_test.go [hunk 2]
```diff
@@ -82,18 +73,14 @@ func (s *DockerSuite) TestNetworkLocalhostTCPNat(c *check.C) {
...
-	if err != nil {
-		c.Fatal(err)
-	}
+	c.Assert(err, check.IsNil)
+
 	final := strings.TrimRight(string(data), "\n")
-	if final != msg {
-		c.Fatalf("Expected message %q but received %q", msg, final)
-	}
+	c.Assert(final, check.Equals, msg, check.Commentf("Expected message %q but received %q", msg, final))
```
Comment (HuKeping) [inline integration-cli/docker_cli_nat_test.go]: The error comment is nothing special so I think c.Assert(final, checker.Equals, msg) is OK.

File: integration-cli/docker_cli_nat_test.go [hunk 3]
```diff
@@ -60,18 +55,14 @@ func (s *DockerSuite) TestNetworkNat(c *check.C) {
...
```
Comment (HuKeping) [inline integration-cli/docker_cli_nat_test.go]: The error comment is nothing special so I think c.Assert(final, checker.Equals, msg) is OK.

File: integration-cli/docker_cli_nat_test.go (hunk 2 above)
Comment (liaoqingwei) [inline integration-cli/docker_cli_nat_test.go]: Yes,but I think the error comment will show the mistake clearly.

Comment (liaoqingwei) [inline integration-cli/docker_cli_nat_test.go]: Yes,but I think the error comment will show the mistake clearly.

Comment (HuKeping) [inline integration-cli/docker_cli_nat_test.go]: That's exactly what the c.Assert is doing :smile:

Comment (vdemeester) [inline integration-cli/docker_cli_nat_test.go]: Agree with @HuKeping, even without the comment it will be ok :)

**Comment by cpuguy83** [comment]
Please make sure to not open multiple PR's for the same change.
Instead you can `git commit -a --amend --no-edit` and then `git push -f origin <branch name>`

File: integration-cli/docker_cli_nat_test.go (hunk 2 above)
Comment (liaoqingwei) [inline integration-cli/docker_cli_nat_test.go]: OK,I will update it.

**Comment by liaoqingwei** [comment]
@cpuguy83, OK, I got it.I will update it.

File: integration-cli/docker_cli_nat_test.go [hunk 4]
```diff
@@ -30,14 +29,11 @@ func getExternalAddress(c *check.C) net.IP {
 	}
 
 	ifaceAddrs, err := iface.Addrs()
-	if err != nil || len(ifaceAddrs) == 0 {
-		c.Fatalf("Error retrieving addresses for eth0: %v (%d addresses)", err, len(ifaceAddrs))
-	}
+	c.Assert(err, check.IsNil)
+	c.Assert(len(ifaceAddrs), check.Not(check.Equals), 0)
```
Comment (vdemeester) [inline integration-cli/docker_cli_nat_test.go]: `c.Assert(ifaceAddrs, checker.Not(checker.HasLen), 0)`

**Comment by vdemeester** [comment]
A nit, othewrise LGTM

**Comment by cpuguy83** [comment]
LGTM after @vdemeester's nit.

File: integration-cli/docker_cli_nat_test.go (hunk 4 above)
Comment (liaoqingwei) [inline integration-cli/docker_cli_nat_test.go]: Yes,it is better actually.Thanks @vdemeester @cpuguy83

**Pull request merged**